### PR TITLE
Use bleeding-edge pyam in CI + reporting hotfix

### PR DESCRIPTION
**Pull request by gidden**
No-op to check if CI is failing on current master

### CODE REVIEW

**Comment by gidden** [comment]
to note, tests are currently failing due to know reporter test errors. @khaeru is investigating

**Comment by gidden** [comment]
@khaeru and/or @danielhuppmann this now fixes our pyam-related CI and hotfixes reporting code based on recent ixmp changes

**Comment by gidden** [comment]
@khaeru should be good to go

**Comment by khaeru** [comment]
Tests will now fail until iiasa/ixmp#178 is merged.

File: message_ix/reporting/__init__.py [hunk 1]
```diff
@@ -179,9 +193,10 @@ def as_pyam(self, quantities, year_time_dim, key=None, drop={},
             Keys for the reporting targets that create the IamDataFrames
             corresponding to *quantities*. The keys have the added tag ‘iamc’.
         """
-        # TODO, this should check for any viable container
-        if not isinstance(quantities, list):
+        if isinstance(quantities, (str, Key)):
```
Comment (gidden) [inline message_ix/reporting/__init__.py]: should this be basestring?

File: .travis.yml [hunk 2]
```diff
@@ -20,7 +20,12 @@ before_install:
 install:
   - git clone https://github.com/iiasa/ixmp.git
   - cd ixmp && python setup.py install && cd ..
-  - pip install .[reporting]
+# Upon the next pyam release (>0.2.0), these two lines should be removed.
+  - git clone https://github.com/IAMconsortium/pyam.git
```
Comment (zikolach) [inline .travis.yml]: @gidden would it be better to use `pip install --quiet --no-cache-dir git+https://github.com/IAMconsortium/pyam.git@master`

Comment (khaeru) [reply .travis.yml]: Good suggestion.

**Pull request merged**